crates/codegen/spec/src/snippets.rs

```diff
@@ -51,21 +52,42 @@ impl<'context> Snippets<'context> {
         let production_name = &production.name();
         let (section, topic) = self.grammar.locate_production(production_name);
 
-        let file_name = match production.versions() {
-            None => "unversioned".to_owned(),
-            Some(versions) => {
-                Production::get_exact_version(versions.into_iter(), &version)?.to_string()
-            }
+        let file_name = match production.as_ref() {
+            Production::Scanner { version_map, .. } => match version_map {
+                VersionMap::Unversioned(_) => Some("unversioned".to_owned()),
+                VersionMap::Versioned(versions) => versions
+                    .keys()
+                    .rev()
+                    .find(|v| *v <= version)
+                    .and_then(|v| versions.get(v).unwrap().as_ref().map(|_| v.to_string())),
```

Comment:
Not sure I understand why each of these three are unwrapping the optional value then discarding it. Is this just checking if `versions.contains_key()`?

Reply:
No, checking if the value is None i.e. this is a deleted production in this version. Easiest way is to map it.

Reply:
Because a deleted production has an entry in the version map with a None value